tensorflow/core/framework/kernel_def.proto

```diff
@@ -16,6 +16,12 @@ message KernelDef {
 
   // Type of device this kernel runs on.
   string device_type = 2;
+ 
+  // Sub-type of device this kernel runs on.
+  // For first party devices, subdevice_type is empty(e.g. CPU:).
+  // For PluggableDevices(third party devices), subdevice_type is registered 
+  // by plugin.(e.g. GPU:X_GPU)
```

Comment:
I believe the grammatically correct sentence is: "by plugin (e.g. GPU:X_GPU).".

Reply:
done